File: src/main/java/upbrella/be/store/dto/response/SingleBusinessHourResponse.java
package upbrella.be.store.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalTimeSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import upbrella.be.store.entity.BusinessHour;

import java.time.DayOfWeek;
import java.time.LocalTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SingleBusinessHourResponse {

    private long id;
    private DayOfWeek date;
    @JsonSerialize(using = LocalTimeSerializer.class)
    @JsonDeserialize(using = LocalTimeDeserializer.class)
    @JsonFormat(pattern = "HH:mm")
    private LocalTime openAt;
    @JsonSerialize(using = LocalTimeSerializer.class)
    @JsonDeserialize(using = LocalTimeDeserializer.class)
    @JsonFormat(pattern = "HH:mm")
    private LocalTime closeAt;

    public static SingleBusinessHourResponse createSingleHourResponse(BusinessHour businessHour) {

            return SingleBusinessHourResponse.builder()
                    .id(businessHour.getId())
                    .date(businessHour.getDate())
                    .openAt(businessHour.getOpenAt())
                    .closeAt(businessHour.getCloseAt())
                    .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleBusinessHourResponse.kt
package upbrella.be.store.dto.response

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalTimeSerializer
import java.time.DayOfWeek
import java.time.LocalTime
import upbrella.be.store.entity.BusinessHour

data class SingleBusinessHourResponse(
    val id: Long?,
    val date: DayOfWeek,
    @field:JsonSerialize(using = LocalTimeSerializer::class)
    @field:JsonDeserialize(using = LocalTimeDeserializer::class)
    @field:JsonFormat(pattern = "HH:mm")
    val openAt: LocalTime?,
    @field:JsonSerialize(using = LocalTimeSerializer::class)
    @field:JsonDeserialize(using = LocalTimeDeserializer::class)
    @field:JsonFormat(pattern = "HH:mm")
    val closeAt: LocalTime?
) {
    companion object {
        fun createSingleHourResponse(businessHour: BusinessHour): SingleBusinessHourResponse {
            return SingleBusinessHourResponse(
                id = businessHour.id,
                date = businessHour.date,
                openAt = businessHour.openAt,
                closeAt = businessHour.closeAt
            )
        }
    }
}
